File: ExampleObjC/GiniVisionExampleObjC/NoResultViewController.m
//
//  NoResultViewController.m
//  GiniVisionExampleObjC
//
//  Created by Peter Pult on 16/08/16.
//  Copyright © 2016 Gini. All rights reserved.
//

#import "NoResultViewController.h"

@interface NoResultViewController ()

@property (strong, nonatomic) IBOutlet UIImageView *rotateImageView;

@end

@implementation NoResultViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.rotateImageView.image = [_rotateImageView.image imageWithRenderingMode:UIImageRenderingModeAlwaysTemplate];
}

- (IBAction)retry:(id)sender {
    [self.navigationController popViewControllerAnimated:YES];
}

@end

File: Example/GiniVision/NoResultViewController.swift
//
//  NoResultViewController.swift
//  GiniVision
//
//  Created by Peter Pult on 22/08/2016.
//  Copyright © 2016 Gini. All rights reserved.
//

import UIKit

protocol NoResultsScreenDelegate:class {
    func noResults(viewController: NoResultViewController, didTapRetry:())
}

final class NoResultViewController: UIViewController {
    
    @IBOutlet var rotateImageView: UIImageView!
    weak var delegate:NoResultsScreenDelegate?
    
    override func viewDidLoad() {
        super.viewDidLoad()
        
        rotateImageView.image = rotateImageView.image?.withRenderingMode(.alwaysTemplate)
    }
    
    @IBAction func retry(_ sender: AnyObject) {
        delegate?.noResults(viewController: self, didTapRetry: ())
    }
}
